### PR TITLE
Fix small buffer handling with the Platform Crypto Provider

### LINKED ISSUE #80444: RSA.SignHash does not reliably work with the Platform Crypto Provider
In places we handle "buffer too small" for RSA signing, this does not work with the the Platform Crypto Provider in most cases. For example:

https://github.com/dotnet/runtime/blob/d4907cf8fb4800b4ec96d07cd93be58eb479a2e6/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngCommon.SignVerify.cs#L30-L34

Signing with a too-small buffer with the Platform Crypto Provider returns a different `SECURITY_STATUS`. It returns `TPM_E_PCP_BUFFER_TOO_SMALL` (0x80290406), not `NTE_BUFFER_TOO_SMALL`.

We should look at places where we check for `NTE_BUFFER_TOO_SMALL` and also handle `TPM_E_PCP_BUFFER_TOO_SMALL` as appropriate for the Platform Crypto Provider.

### PR DESCRIPTION
**Pull request by vcsjones**
When using the Platform Crypto Provider with CNG, it returns different a different error code for "buffer too small".

This means for `Try*` APIs we will throw an exception when we should be returning false. For allocating-return APIs, this would mean we throw instead of returning a right sized buffer.

This change allows handling `TPM_E_PCP_BUFFER_TOO_SMALL` in places we were also handling `NTE_BUFFER_TOO_SMALL`.

Fixes #80444.

### CODE REVIEW

File: src/libraries/Common/src/Interop/Windows/NCrypt/Interop.ErrorCode.cs [hunk 1]
```diff
@@ -20,6 +20,12 @@ internal enum ErrorCode : int
             NTE_NO_MORE_ITEMS = unchecked((int)0x8009002a),
             E_FAIL = unchecked((int)0x80004005),
             STATUS_UNSUCCESSFUL = unchecked((int)0xC0000001),
+            TPMAPI_E_BUFFER_TOO_SMALL = unchecked((int)0x80290406),
```
Comment (adamsitnik) [inline src/libraries/Common/src/Interop/Windows/NCrypt/Interop.ErrorCode.cs:23]: According to https://learn.microsoft.com/en-us/windows/win32/com/com-error-codes-6 the value of `TPMAPI_E_BUFFER_TOO_SMALL` is equal to `0x80290106`. I am guessing it's a typo?

```suggestion
            TPMAPI_E_BUFFER_TOO_SMALL = unchecked((int)0x80290106),
```

**Review by adamsitnik — COMMENTED** [review]
Overall it LGTM, but before I hit the approve & merge buttons I would like to make sure that we are using the right `TPMAPI_E_BUFFER_TOO_SMALL` value.

Thank you for your contribution @vcsjones !

File: src/libraries/Common/src/Interop/Windows/NCrypt/Interop.ErrorCode.cs (hunk 1 above)
Comment (vcsjones) [reply src/libraries/Common/src/Interop/Windows/NCrypt/Interop.ErrorCode.cs]: Ah I used the right value but the wrong name. It should be `TPM_E_PCP_BUFFER_TOO_SMALL`. Thanks for checking!

**Review by adamsitnik — APPROVED** [review]
:shipit:

File: src/libraries/Common/src/Interop/Windows/NCrypt/Interop.ErrorCode.cs [hunk 2]
```diff
@@ -20,6 +20,12 @@ internal enum ErrorCode : int
             NTE_NO_MORE_ITEMS = unchecked((int)0x8009002a),
             E_FAIL = unchecked((int)0x80004005),
             STATUS_UNSUCCESSFUL = unchecked((int)0xC0000001),
+            TPM_E_PCP_BUFFER_TOO_SMALL = unchecked((int)0x80290406),
```
Comment (stephentoub) [inline src/libraries/Common/src/Interop/Windows/NCrypt/Interop.ErrorCode.cs:23]: nit: the enum appears to be sorted by value (except for E_FAIL)... maybe this should be moved up a line?

**Review by stephentoub — APPROVED** [review]

File: src/libraries/Common/src/Interop/Windows/NCrypt/Interop.ErrorCode.cs (hunk 2 above)
Comment (vcsjones) [reply src/libraries/Common/src/Interop/Windows/NCrypt/Interop.ErrorCode.cs:22]: I value sorted the enum (moved the new one and `E_FAIL`).

**Pull request merged**